app/main.py

```diff
@@ -1 +1,46 @@
-#  write your code here
+
+class Animal:
+    def __init__(self, name: str,
+                 appetite: int,
+                 is_hungry : bool = True) -> None:
+        self.name = name
+        self.appetite = appetite
+        self.is_hungry = is_hungry
+
+    def print_name(self) -> None:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self) -> int:
+        if self.is_hungry:
+            self.is_hungry = False
+            print(f"Eating {self.appetite} food points...")
+            return self.appetite
+        return 0
+
+
+class Cat(Animal):
+    def __init__(self, name : str,
+                 is_hungry : bool = True) -> None:
+        super().__init__(name, appetite=3, is_hungry=is_hungry)
+
+    @staticmethod
+    def catch_mouse() -> None:
+        print("The hunt began!")
+
+
+class Dog(Animal):
+    def __init__(self, name : str,
+                 is_hungry : bool = True) -> None:
+        super().__init__(name, appetite=7, is_hungry=True)
```

Comment:
In the `Dog` class constructor, the `is_hungry` parameter is passed as `True` directly to the superclass constructor, ignoring the `is_hungry` argument provided to the `Dog` constructor. This means that the `is_hungry` parameter in the `Dog` constructor is redundant and should be used instead of hardcoding `True`.